libraries/common-arm/src/lib.rs

```diff
@@ -1 +1,161 @@
 #![no_std]
+use core::marker::PhantomData;
```

Comment:
Could you move all of this to its own module? Take a look at my error management PR if you need an example.

Reply:
I've moved this over to its own module with the help of your error management PR.